unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneResourcesLoadTracker/ResourcesLoadTrackerECS.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using DCL;
+using DCL.WorldRuntime;
+using UnityEngine;
+
+public class ResourcesLoadTrackerECS : IResourcesLoadTracker
+{
+    public int pendingResourcesCount => resourcesNotReady.Count;
+    
+    public float loadingProgress 
+    {
+        get
+        {
+            int resourcesReadyCount = resourcesReady.Count;
+            return resourcesReadyCount > 0 ? (resourcesReadyCount - pendingResourcesCount) * 100f / resourcesReadyCount : 100f;
+        }
+    }
+    
+    public event Action OnResourcesLoaded;
+    public event Action OnStatusUpdate;
+    
+    private readonly List<object> resourcesNotReady = new List<object>();
+    private readonly List<object> resourcesReady = new List<object>();
```

Comment:
Do we need to know the ready resources? what's the need?

This class might even dissapear if we just use a Dictionary as I mentioned above to track all the resources per scene.

Reply:
I added it for debugging reasons, since you can always check which models are loading and which one not, but I will change it for a int